### PR TITLE
Feature cognito s3 stagiaire

### PR DESCRIPTION
**Pull request by CedricJean**
It's a feature who enable the app to work with the storage of Amazon or AWS.

With the actual progress the feature support four type of project :
- Empty
- Image Classification
- Image Segmentation
- Video Segmentation

Interface of import :

tempsnip

Interface for connection :

capture

Interface for loading a project :

image

It save automatically when the user change the data

### CODE REVIEW

**Comment by seveibar** [comment]
tagging @puskuruk to take a look as well. Thanks for all the screenshots @CedricJean!

File: desktop/main.js [hunk 1]
```diff
@@ -5,7 +5,6 @@ const menuTemplate = require("./menu-template")
 const { format: formatUrl } = require("url")
 
 function createWindow() {
-  console.log("Creating window...")
```
Comment (puskuruk) [inline desktop/main.js:8]: :smile:

File: src/components/LocalStorageApp/index.js [hunk 2]
```diff
@@ -78,6 +112,96 @@ export default () => {
...
+          !changes.content.taskOutput &&
+          !changes.fileName) ||
+        (file.content.interface.type !== "video_segmentation" &&
+          file.content.interface.type !== "image_classification" &&
+          file.content.interface.type !== "image_segmentation")
+      )
+        return
+      lastObjectRef.current = file
+      UpdateAWSStorage()
+    }
```
Comment (seveibar) [inline src/components/LocalStorageApp/index.js:202]: too much auth-specific code in OHAEditor :) can you isolate this to something inside of `file-handlers`?

File: src/components/ImportPage/index.js [hunk 3]
```diff
@@ -148,24 +222,129 @@ export default ({ oha, onChangeOHA, isDesktop }) => {
...
-            appendedTaskOutput
-          )
+          ["taskData"],
+          (oha.taskData || []).concat(json.content.taskData)
+        )
+        newOHA = setIn(newOHA, ["interface"], json.content.interface)
+        if (typeof file.fileName === "undefined" || file.fileName === "unnamed")
+          file = setIn(file, ["fileName"], json.fileName)
+        file = setIn(file, ["content"], newOHA)
+        onChangeFile(file, true)
```
Comment (seveibar) [inline src/components/ImportPage/index.js:337]: gah we need `onChangeFile` just to change the file name :|

I don't see a way around this for now.

File: src/utils/RecognizeFileExtension.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    fileExtension === "jpeg" ||
+    fileExtension === "png" ||
+    fileExtension === "ico" ||
+    fileExtension === "jpe" ||
+    fileExtension === "gif"
+  )
+    typeOfFile = "Image"
+  if (fileExtension === "mp4" || fileExtension === "mkv") typeOfFile = "Video"
+  if (fileExtension === "mp3") typeOfFile = "Audio"
+  return typeOfFile
```
Comment (seveibar) [inline src/utils/RecognizeFileExtension.js:17]: nice

File: src/utils/fileHasChanged.js [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+
+  // Vérifie si la donnée fileName existe et est changé
+  if (typeof objectToCheck.fileName !== "undefined") {
+    if (typeof objectOfRef.fileName === "undefined") resultSet.fileName = true
+    else if (objectToCheck.fileName !== objectOfRef.fileName)
+      resultSet.fileName = true
+  }
+  //Comportement par défaut
+  return resultSet
+}
```
Comment (seveibar) [inline src/utils/fileHasChanged.js:140]: don't forget about poor mono-lingual people like me :)

File: src/components/Composite/index.js [hunk 6]
```diff
@@ -33,7 +33,6 @@ export const Composite = (props) => {
     interface: { fields },
   } = props
   const [selectedField, changeSelectedField] = useState()
-  const [taskOutput, changeTaskOutput] = useState(props.taskOutput)
```
[the remark this answers is not in the document]
Comment (CedricJean) [reply src/components/Composite/index.js:36]: In fact this var was never used

File: src/components/ImportPage/index.js [hunk 7]
```diff
@@ -148,24 +222,129 @@ export default ({ oha, onChangeOHA, isDesktop }) => {
...
+          v++
+        } else {
+          appendedTaskData[i].sampleName = sampleName[1]
+          boolName = false
+        }
+      }
+      appendedTaskData[i].sampleName = sampleName[1]
+    }
+    return appendedTaskData
+  }
```
[the remark this answers is not in the document]
Comment (CedricJean) [reply src/components/ImportPage/index.js:310]: Yep, its a way to deal with two file with the same name when these two files is different. I check the origin(the path) to determine if those file is different

File: src/components/ImportPage/index.js (hunk 3 above)
Comment (CedricJean) [reply src/components/ImportPage/index.js:337]: Okay if you have a better idea feel free to ask. I will change it

File: src/utils/fileHasChanged.js (hunk 5 above)
Comment (CedricJean) [reply src/utils/fileHasChanged.js:140]: I will change the comment in english

File: src/components/HeaderToolbar/index.js [hunk 8]
```diff
@@ -132,6 +135,21 @@ const HeaderToolbar = ({
...
+            }}
+            className={c.headerButton}
+          >
+            Login
+          </Button>
+        )}
+        {!isEmpty(authConfig) && !isEmpty(user) && (
+          <Button onClick={logoutUser} className={c.headerButton}>
+            Logout
+          </Button>
```
[the remark this answers is not in the document]
Comment (Ownmarc) [reply src/components/HeaderToolbar/index.js:151]: login/logout wont show up if auth is not configured.

File: src/utils/fileHasChanged.js (hunk 5 above)
Comment (seveibar) [reply src/utils/fileHasChanged.js:141]: cheers :)

**Comment by seveibar** [comment]
Thanks for making those changes @CedricJean, code looks good to me, running some tests now.

I think after we merge this we should get a some credentials for the CI and a fixed sort of setup for continuous integration testing (maybe we could add a Cypress test), we want to make sure we don't break it :)

**Comment by CedricJean** [comment]
I can't merge its say a change is requested but i didn't find it

**Comment by seveibar** [comment]
yep it required me to re-review :)

**Comment by seveibar** [comment]
so my fault

**Comment by seveibar** [comment]
@all-contributors Please add @Ownmarc for code and docs

**Pull request merged**